src/com/google/javascript/jscomp/GlobalNamespace.java

```diff
@@ -1123,7 +1149,7 @@ boolean shouldKeepKeys() {
     }
 
     boolean needsToBeStubbed() {
-      return globalSets == 0 && localSets > 0;
+      return globalSets == 0 && localSets > 0 && localSetsWithNoCollapse <= 0;
```

Comment:
In the case where the object is global, but the property is assigned in a local scope, this is needed.

See https://github.com/google/closure-compiler/pull/805/files#diff-f6ebc91c417a43a65475b9a83521bc54R855